public/app/features/correlations/Forms/QueryEditorField.tsx

```diff
@@ -13,6 +24,19 @@ interface Props {
 }
 
 export const QueryEditorField = ({ dsUid, invalid, error, name }: Props) => {
+  const [isValidQuery, setIsValidQuery] = useState<boolean | undefined>(undefined);
```

Comment:
wouldn't `isQueryValid` be a better name instead?

Reply:
`isQueryValid` sounds to me like it is a question. Can you please explain a bit more why it would be better in your opinion?

Reply:
that's usually how those boolean variables are read imho

in `isQueryValid` being `Valid` a classifying adjective to `Query`, it completely describes the variable state, which is going to be `true` if the query is valid. "is Query valid?" is a complete sentence in english.

`isValidQuery` on the other hand lacks an adjective. "is valid Query?" is not a complete sentence in english, and it lacks and adjective to be complete as - in this form - "valid query" is the subject of the question.

the rule i usually follow is to name boolean variables and functions returning booleans as `is[subject][classifier]` and functions as `is[classifier](subject)`. i.e. `isWindowOpen` and `isOpen(window)`

in other words, what we are interested in is if the query is valid (isQueryValid), not if something is a valid query. if `isValidQuery` was a function i would have agreed with you when used like `isValidQuery(myQuery)`. 

anyway this was more a nit than anything else, feel free to ignore.

Reply:
Responding here after a DM from @Elfo404 to check with a technical writer; my response was:

"`isQueryValid` says to me, "Is the | this query valid?"
"`isValidQuery` says to me, "[It] is a valid query." or "The | This query is valid."

Reply:
I prefer to leave it as it is.